clang/lib/Parse/ParseExpr.cpp

```diff
@@ -1065,6 +1065,18 @@ ExprResult Parser::ParseCastExpression(CastParseKind ParseKind,
     break;
   }
 
+  case tok::annot_embed: {
+    // We've met #embed in a context where a single value is expected. Take last
+    // element from #embed data as if it were a comma expression.
+    EmbedAnnotationData *Data =
+        reinterpret_cast<EmbedAnnotationData *>(Tok.getAnnotationValue());
+    SourceLocation StartLoc = ConsumeAnnotationToken();
+    ASTContext &Context = Actions.getASTContext();
+    return IntegerLiteral::Create(
+        Context, llvm::APInt(CHAR_BIT, Data->BinaryData.back()),
+        Context.UnsignedCharTy, StartLoc);
+  } break;
+
```

Comment:
> What if the embed is not empty? We should diagnose!

Do you mean if **it is empty**? 

> (alternatively, I this we should inject back individual literals into the token streams)

Can we actually create new tokens here and put them back here in Parser code?
Is there an example of how this can be done?

Reply:
> Do you mean if it is empty?

Sorry, If it has more than 1 elements

> Can we actually create new tokens here and put them back here in Parser code?
Is there an example of how this can be done?

Look for calls to `PP.EnterTokenStream` in the Parser

`PragmaSTDC_FENV_ROUNDHandler::HandlePragma` is a good example, I think

Reply:
Test case:
```
$ printf '\x01\x02' > hello
$ printf 'char f(char *buf) {\nreturn \n#embed "hello"\n[buf];\n}\n' > hello.c
$ build/bin/clang -S -o - -std=c23 hello.c
hello.c:3:2: error: expected ';' after return statement
    3 | #embed "hello"
      |  ^
      |  ;
```

This ought to create the statement `return 1, 2[buf];`, which is valid -- though the LHS is useless. It should be warned about in the usual fashion, e.g. `warning: left operand of comma operator has no effect [-Wunused-value]`.

Reply:
Fixed bug and warning.